audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -38,6 +46,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        //get run-time permission...
+        requestResourcePermission();
```

Comment:
yes, that's true.

I was mainly wondering if it would be confusing since we can't display any 'messages' to the user before we show the permission.